app/workers/set_tenant_id_worker.rb

```diff
@@ -0,0 +1,18 @@
+class SetTenantIdWorker < ApplicationJob
+
+  def perform(*)
+    ActiveRecord::Base.transaction do
+      Account.where("provider = 1 AND master IS NULL").update_all(master: false)
```

Comment:
`update_all` will lock the `accounts` table for as long as it is executed. This is dangerous as we have a live application:

1. If a request updates the data of an Account, it will wait until the lock is released or raise a Lock wait timeout exceeded error.
2. That error will return to the user as a 5xx which is not nice
3. it will slow down the application, eventually leading to a global Denial of Service

Some advices, make it simpler at the beginning, you can probably make it more optimized later but will you need it?

1. enqueue per provider the update
2. in the `SetTenantId#perform` do the update for the provider and the other models that depend on it but per object, having a loop is not bad, we can iterate from there

Reply:
Thanks @hallelujah I changed the `update_all` for smaller updates of batches of 100. That should be small enough to avoid significant locks. Wdyt?
```ruby
loop do
  ids = Account.where("provider = 1 AND master IS NULL").limit(100).to_a.map { |account| account.id }
  break if ids.empty?

  Account.where(id: ids).update_all(master: false)
end
```